src/supervisor/policyAnalysis.py

```diff
@@ -333,42 +343,42 @@ def getRootPolicyControllerResponseTimeforSpecChange(pc,startTime, endTime, step
-        propagated_policy_number_trend_df.plot(title="Percentage of root-policy-spec reconcile times le 1 se",figsize=(30, 15))
-        plt.savefig('../../output/policy-root-ctrl-spec-le1sec-count.png')
-        saveCSV(propagated_policy_number_trend_df[['RootPolicySpecCtrlResponsePctle1sec']],"policy-root-ctrl-spec-le1sec-count",True)
-        plt.savefig('../../output/breakdown/policy-root-ctrl-spec-le1sec-count.png')
-        saveCSV(propagated_policy_number_trend_df,"policy-root-ctrl-spec-le1sec-count")
+        propagated_policy_number_trend_df.rename(columns={"value": "RootPolicySpecCtrlResponse95Pctle"}, inplace = True)
+        propagated_policy_number_trend_df.plot(title="95th Percentile root-policy-spec reconcile times le 1 se",figsize=(30, 15))
+        plt.savefig('../../output/policy-root-ctrl-spec-95pctle.png')
+        saveCSV(propagated_policy_number_trend_df[['RootPolicySpecCtrlResponse95Pctle']],"policy-root-ctrl-spec-le1sec-count",True)
+        plt.savefig('../../output/breakdown/policy-root-ctrl-spec-95pctle.png')
+        saveCSV(propagated_policy_number_trend_df,"policy-root-ctrl-spec-95pctle")
         plt.close('all') 
 
     except Exception as e:
-        print(Fore.RED+"Error in getting Percentage of root-policy-spec reconcile times le 1 sec: ",e)
+        print(Fore.RED+"Error in getting 95th Percentile root-policy-spec reconcile times: ",e)
         print(Style.RESET_ALL)    
     print("=============================================")
    
     status=True
     return status
 
 def getRootPolicyControllerResponseTimeforStatusChange(pc,startTime, endTime, step):
-    print("Get Percentage of root-policy-status reconcile times le 1 sec")
+    print("Get 95th Percentile root-policy-status reconcile times")
 
     # This metric should be strongly causally related to getReplicatedPolicyStatusChange
 
     # This should have got us the 95th percentike
     # histogram_quantile(0.95, sum(rate(controller_runtime_reconcile_time_seconds_bucket{controller="replicated-policy"}[10m])) by (le))
     # however, if no policies are to be replicated, this metric does not change. Therefore we do not have results for this period
-    # Thus for this type of behaviour, we cannot use this technique
+    # Thus we use: histogram_quantile(0.95, sum(rate(controller_runtime_reconcile_time_seconds_bucket{controller="root-policy-status"}[30m])) by (le)) or vector(0)
     try:
-        propagated_policy = pc.custom_query('sum without(le,instance,pod) (controller_runtime_reconcile_time_seconds_bucket{controller="root-policy-status",le="1"})/sum without(le,instance,pod) (controller_runtime_reconcile_time_seconds_count{controller="root-policy-status"})')
+        propagated_policy = pc.custom_query('histogram_quantile(0.95, sum(rate(controller_runtime_reconcile_time_seconds_bucket{controller="root-policy-status"}[30m])) by (le)) or vector(0)')
```

Comment:
If I got a `NaN` value does it just mean no data in 30 minute window?

Reply:
In other words, this metric loses it's value unless it's captured within the 30 minute window of hitting the problem.  So I'm not sure if this is the right way to analyze the data during an acm-inspector run

Reply:
This query histogram_quantile(0.95, sum(rate(controller_runtime_reconcile_time_seconds_bucket{controller="root-policy-status"}[30m])) by (le)) or vector(0) should never return NaN because of vector(0). This means when there is no data in 30 min interval, we should get 0.  @saswatamcode I was looking at the detailed output. Looks like `or vector(0)` is not replacing NaN by a 0 ?

Reply:
Sorry for the delay on this. Ok so I don't see any issues with the query itself and this is an instant query anyway, so histogram_quantile would always return some value or single NaN which the vector(0) default can handle.

Could you point me to where you are getting NaNs with this query?

In theory, if the histogram metric doesn't have the correct buckets or some have missing data/gaps, histogram_quantile may return a NaN. If this were a range query, it would be tricky to filter these out, as vector(0) or would apply over it. But for instant query like here, this should work fine I think

Reply:
Well - whatever it is vector(0) does not seem to cover it. Now, this can produce complication in the results - which are subsequently analyzed. So added a suppression on the pandas side to fill nan with a 0. This should be technically fine in the semantic sense as well for this data. @gparvin @saswatamcode